src/action-creators/undoArchive.js

```diff
@@ -1,12 +1,25 @@
 import { store } from '../store'
 
+// utils
+import {
+  contextOf,
+  getThoughtsRanked,
+  head,
+  pathToContext,
+  rankThoughtsFirstMatch,
+  splice,
+} from '../util'
+
 // action-creators
 import alert from '../action-creators/alert'
 
 export const undoArchive = ({ originalPath, currPath, offset }) => dispatch => {
 
   const state = store.getState()
 
+  const archiveContext = contextOf(pathToContext(currPath))
+  const contextOfArchive = splice(archiveContext, archiveContext.length - 1, 1)
```

Comment:
Any reason you are using `splice` instead of `contextOf` here?

Reply:
The names `archiveContext` and `contextOfArchive` seem to signify the same thing. Can we clarify these names so that they are semantically different? I would suggest:

Rename `archiveContext` → `context`
Rename `contextOfArchive` → `archiveContext`

Reply:
`splice` was used to build the context of archive, but it can also be retrieved from `originalPath`, which is being used now.